exporter/datadogexporter/traces_exporter.go

```diff
@@ -32,11 +33,27 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/utils"
 )
 
+// sublayerCalculator is thread safe wrapper of a sublayer
+// calculator. Each trace exporter has a single sublayer
+// calculator that is reused by each push
+type sublayerCalculator struct {
+	sc    *stats.SublayerCalculator
+	mutex sync.Mutex
+}
+
+// ComputeSublayers computes the sublayers of a trace
+func (s *sublayerCalculator) ComputeSublayers(trace pb.Trace) []stats.SublayerValue {
```

Comment:
Why public?

Reply:
See reply in the other comment, marking as solved.